src/Microsoft.ApplicationInsights.AspNet/Extensibility/Implementation/Tracing/TelemetryLogger.cs

```diff
@@ -0,0 +1,120 @@
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.ApplicationInsights.AspNet.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Diagnostics.Tracing;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// Application Insights internal telemetry logging implementation.
+    /// </summary>
+    internal sealed class TelemetryLogger : ILogger
+    {
+        /// <summary>Instance of </summary>
+        private static TelemetryLogger instance = new TelemetryLogger();
+
+        /// <summary>The underlying event source that internal telemetry is logged to.</summary>
+        private AspNetEventSource eventSource;
+
+        /// <summary>Lookup table of event source log calls indexed by LogLevel.</summary>
+        private EventSourceLogCall[] eventSourceLogCalls;
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryLogger class.
+        /// </summary>
+        public TelemetryLogger() : this(LogLevel.Error)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryLogger class.
+        /// </summary>
+        /// <param name="level">The logging level.</param>
+        public TelemetryLogger(LogLevel level)
+        {
+            this.Level = level;
+            this.eventSource = AspNetEventSource.Instance;
+
+            // The event levels are reordered for indexing by the log levels, this is a quick but potentially brittle way to do translation.
+            this.eventSourceLogCalls = new EventSourceLogCall[]
+            {
+                null,
+                this.eventSource.LogAlways,        // EventLevel.LogAlways = 0     | LogLevel.Debug = 1
+                this.eventSource.LogVerbose,       // EventLevel.Verbose = 5       | LogLevel.Verbose = 2
+                this.eventSource.LogInformational, // EventLevel.Informational = 4 | LogLevel.Information = 3
+                this.eventSource.LogWarning,       // EventLevel.Warning = 3       | LogLevel.Warning = 4
+                this.eventSource.LogError,         // EventLevel.Error = 2         | LogLevel.Error = 5
+                this.eventSource.LogCritical,      // EventLevel.Critical = 1      | LogLevel.Critical = 6
+            };
+        }
+
+        /// <summary>
+        /// Delegate of basic event source function calls.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        /// <param name="appDomainName">An ignored placeholder to make EventSource happy.</param>
+        private delegate void EventSourceLogCall(string message, string appDomainName = "Incorrect");
+
+        /// <summary>
+        /// Gets an instance of the telemetry logger.
+        /// </summary>
+        public static TelemetryLogger Instance
```

Comment:
nit-pick: "Instance" seems to be used in event source, and in this code as well.